e2e-test/log/src/integrationTest/java/com/amazon/dataprepper/integration/log/EndToEndBasicGrokTest.java

```diff
@@ -0,0 +1,139 @@
+                    Assert.assertEquals(5, foundSources.size());
+                    retrievedDocs.addAll(foundSources);
+                }
+        );
+        // Verify original and grokked keys from retrieved docs
+        final List<String> expectedDocKeys = Arrays.asList(
+                "date", "log", "clientip", "ident", "auth", "timestamp", "verb", "request", "httpversion", "response", "bytes");
+        retrievedDocs.forEach(expectedDoc -> {
+            for (String key: expectedDocKeys) {
+                Assert.assertNotNull(expectedDoc.get(key));
+            }
+        });
+    }
+
+    private RestHighLevelClient prepareOpenSearchRestHighLevelClient() {
+        final ConnectionConfiguration.Builder builder = new ConnectionConfiguration.Builder(
+                Collections.singletonList("https://127.0.0.1:9200"));
+        builder.withUsername("admin");
+        builder.withPassword("admin");
+        return builder.build().createClient();
+    }
+
+    private void sendHttpRequestToSource(final int port, final HttpData httpData) {
+        WebClient.of().execute(RequestHeaders.builder()
+                        .scheme(SessionProtocol.HTTP)
+                        .authority(String.format("127.0.0.1:%d", port))
+                        .method(HttpMethod.POST)
+                        .path("/log/ingest")
+                        .contentType(MediaType.JSON_UTF_8)
+                        .build(),
+                httpData)
+                .aggregate()
+                .whenComplete((i, ex) -> assertThat(i.status(), is(HttpStatus.OK))).join();
+    }
+
+    private List<Map<String, Object>> getSourcesFromSearchHits(final SearchHits searchHits) {
+        final List<Map<String, Object>> sources = new ArrayList<>();
+        searchHits.forEach(hit -> {
+            Map<String, Object> source = hit.getSourceAsMap();
+            sources.add(source);
+        });
+        return sources;
+    }
+
+    private void refreshIndices(final RestHighLevelClient restHighLevelClient) throws IOException {
+        final RefreshRequest requestAll = new RefreshRequest();
+        restHighLevelClient.indices().refresh(requestAll, RequestOptions.DEFAULT);
+    }
+
+    private HttpData generateRandomApacheLogHttpData(final int numLogs) throws JsonProcessingException {
+        final List<Map<String, Object>> jsonArray = new ArrayList<>();
+        for (int i = 0; i < numLogs; i++) {
+            final Map<String, Object> logObj = new HashMap<String, Object>() {{
+                put("date", System.currentTimeMillis());
+                put("log", apacheLogFaker.generateRandomApacheLog());
+            }};
+            jsonArray.add(logObj);
+        }
+        final String jsonData = objectMapper.writeValueAsString(jsonArray);
+        return HttpData.ofUtf8(jsonData);
+    }
+
+    public static void main(String[] args) throws JsonProcessingException {
```

Comment:
I'm not sure I understand why we need this `main` method here with the `print` statement?

Reply:
Ah. That is my mis. Great catch! Will remove that.